solr/core/src/test/org/apache/solr/client/solrj/embedded/TestEmbeddedSolrServerConstructors.java

```diff
@@ -58,4 +63,63 @@ public void testNodeConfigConstructor() throws Exception {
       assertEquals(1, server.query("newcore", new SolrQuery("*:*")).getResults().getNumFound());
     }
   }
+
+  @SuppressWarnings("removal")
+  @Test
+  public void testPathConstructorZipFS() throws Exception {
+    assumeTrue(
+        """
+                    Test only works without Security Manager due to SecurityConfHandlerLocal
+                    missing permission to read /1/2/3/4/security.json file""",
+        System.getSecurityManager() == null);
```

Comment:
To make the test work regardless of the security manager, this line https://github.com/apache/solr/blob/a5b0b98a79559021e44d00b43fff6c020c54f829/solr/core/src/java/org/apache/solr/handler/admin/SecurityConfHandlerLocal.java#L43 needs to be updated to handle Path in a more idiomatic way.

However, that requires updating CoreContainer#getSolrHome to return a Path instead of a String - so it looks like a much bigger change.

Reply:
CoreContainer.solrHome is a Path so hopefully it won't be too huge to switch it's getter to use Path.

Reply:
OK, I think I will keep this PR as draft for now, and work on updating CoreContainer#getSolrHome to return Path first. Is there an existing JIRA for that?

Reply:
Removed the assumption about the SM now that #3204 is merged - the tests should be able to pass now.